backend/service/k8s/deployment.go

```diff
@@ -236,3 +303,72 @@ func updateContainerResources(deployment *appsv1.Deployment, fields *k8sapiv1.Up
 	}
 	return nil
 }
+
+func updateContainerProbes(deployment *appsv1.Deployment, fields *k8sapiv1.UpdateDeploymentRequest_Fields) error {
```

Comment:
I wonder if it's possible to simplify this code using something like https://github.com/kubernetes/apimachinery/blob/master/pkg/util/strategicpatch/patch.go